hazelcast/src/main/java/com/hazelcast/config/MultiMapConfig.java

```diff
@@ -382,21 +392,21 @@ public void readData(ObjectDataInput in) throws IOException {
         statisticsEnabled = in.readBoolean();
         if (in.getVersion().isGreaterOrEqual(Versions.V3_10)) {
             quorumName = in.readUTF();
+            mergePolicyConfig = in.readObject();
         }
     }
 
     @Override
     @SuppressWarnings({"checkstyle:cyclomaticcomplexity", "checkstyle:npathcomplexity"})
-    public boolean equals(Object o) {
+    public final boolean equals(Object o) {
```

Comment:
I've re-recreated the `equals()` and `hashCode()` methods via IntelliJ, now the test for these methods is passing.